src/checker.rs

```diff
@@ -47,17 +47,34 @@ pub fn check(rules: &Rules, raw: &str) -> bool {
         return false;
     }
 
-    let mut words = trimmed.split_whitespace();
+    let words = trimmed.split_whitespace();
     let word_count = words.clone().count();
     if word_count < rules.min_word_count
         || word_count > rules.max_word_count
-        || words.any(|word| rules.disallowed_words.contains(
+        || words.clone().any(|word| rules.disallowed_words.contains(
              &word.trim_matches(|c: char| !c.is_alphabetic()).to_lowercase()
            ))
     {
         return false;
     }
 
+    if !rules.stem_separator_regex.is_empty() {
+        let regex: Regex = Regex::new(&rules.stem_separator_regex).unwrap();
+        let mut stems_words: Vec<&str> = vec![];
+        
+        for word in words {
+            let maybe_stem_word: &str = &(regex.split(word).nth(0).unwrap_or(word));
+            if maybe_stem_word != word {
+                stems_words.push(&maybe_stem_word);
+            }
+        }
+
+        if stems_words.into_iter().any(|word| rules.disallowed_words.contains(word))
+        {
```

Comment:
very minor and really only because you will need to adjust it anyway: can you move the `{` brace to the if line as well?